internal/common/baseclient/iito.go

```diff
@@ -2,43 +2,61 @@ package baseclient
 
 import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 
-// RequestGift allows clients to signalize that they want a gift
+// GetGiftRequests allows clients to signalize that they want a gift
 // This information is fed to OfferGifts of all other clients.
 // COMPULSORY, you need to implement this method
-func (c *BaseClient) RequestGift() uint {
-	if c.clientGameState.ClientInfo.LifeStatus == shared.Critical {
-		return 100
+func (c *BaseClient) GetGiftRequests() []shared.GiftRequest {
+	requests := []shared.GiftRequest{}
+
+	// FIXME: Is this the right way to get the client statuses?
+	for team, status := range c.clientGameState.ClientLifeStatuses {
+		if status == shared.Critical {
+			requests[team] = shared.GiftRequest{RequestFrom: shared.ClientID(team), RequestAmount: 100.0}
+		} else {
+			requests[team] = shared.GiftRequest{RequestFrom: shared.ClientID(team), RequestAmount: 0.0}
+		}
 	}
-	return 0
+	return requests
 }
 
-// OfferGifts allows clients to offer to give the gifts requested by other clients.
-// It can offer multiple partial gifts
-// COMPULSORY, you need to implement this method
-func (c *BaseClient) OfferGifts(giftRequestDict shared.GiftDict) (shared.GiftDict, error) {
-	return giftRequestDict, nil
+// GetGiftOffers allows clients to make offers in response to gift requests by other clients.
+// It can offer multiple partial gifts.
+// COMPULSORY, you need to implement this method. This placeholder implementation offers no gifts.
+func (c *BaseClient) GetGiftOffers(receivedRequests []shared.GiftRequest) ([]shared.GiftOffer, error) {
+	offers := []shared.GiftOffer{}
+
+	// FIXME: Or is this the right way?
+	clientStatuses := c.serverReadHandle.GetGameState().ClientLifeStatuses
+	for toTeam, status := range clientStatuses {
+		if status == shared.Critical {
+			offers[toTeam] = shared.GiftOffer{ReceivingTeam: shared.ClientID(toTeam), OfferAmount: 100.0}
+		} else {
+			offers[toTeam] = shared.GiftOffer{ReceivingTeam: shared.ClientID(toTeam), OfferAmount: 0.0}
+		}
+	}
+	return offers, nil
```

Comment:
Same issue: please review @lhl2617 @mpardalos

Reply:
This one looks good